ambry-commons/src/main/java/com/github/ambry/commons/AmbryCache.java

```diff
@@ -54,12 +56,15 @@ public class AmbryCache {
    * @param cacheEnabled Toggles cache. If true, cache is enabled. Else, cache is disabled.
    * @param cacheMaxSizeBytes Maximum memory footprint of the cache
    * @param metricRegistry Instance of metrics registry to record stats
+   * @param notificationSystem
    */
-  public AmbryCache(String cacheId, boolean cacheEnabled, long cacheMaxSizeBytes, MetricRegistry metricRegistry) {
+  public AmbryCache(String cacheId, boolean cacheEnabled, long cacheMaxSizeBytes, MetricRegistry metricRegistry,
```

Comment:
On second thought, I think I found a way to do it without notifications. I'll implement and publish.

Reply:
The point is to not use NotificationSystem, since NotificationSystem was created for a specific reason and adding cache hit and miss event to NotificationSystem is not good idea. Let's make sure each class follow the true purpose of their creation. 
That being said, I don't oppose passing a "CacheEventCallback" or something like that, as long as it's not NotificationSystem.